src/Services/HubspotCommerce.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Drupal\hubspot_commerce\Services;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\hubspot_api\Manager;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Exception;
+
+/**
+ * The HubspotCommerce service class.
+ *
+ * Contains functions to sync data with the Hubspot API.
+ *
+ * @package Drupal\hubspot_commerce
+ */
+class HubspotCommerce {
+
+  /**
+   * The Hubspot API Manager.
+   *
+   * @var \Drupal\hubspot_api\Manager
+   */
+  protected $hubspotManager;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Logger service.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * The client.
+   *
+   * @var \SevenShores\Hubspot\Http\Client
+   */
+  public $client;
+
+  /**
+   * Constructs a new HubSpot Commerce service instance.
+   *
+   * @param \Drupal\hubspot_api\Manager $hubspot_manager
+   *   The Hubspot API Manager class.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manger.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger_factory
+   *   The logger factory.
+   *
+   * @throws \Exception
+   */
+  public function __construct(
+    Manager $hubspot_manager,
+    EntityTypeManagerInterface $entity_type_manager,
+    LoggerChannelFactoryInterface $logger_factory
+  ) {
+    $this->hubspotManager = $hubspot_manager;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->logger = $logger_factory->get(HUBSPOT_COMMERCE_LOGGER_CHANNEL);
+
+    // Initialize our Hubspot API client.
+    try {
+      $this->client = $this->hubspotManager->getHandler()->client;
+    }
+    catch (Exception $e) {
+      $this->logger->error('Could not create a Hubspot API client. Error: @error', [
+        '@error' => $e->getMessage(),
+      ]);
```

Comment:
**CHANGE REQUEST**

Let's respect line length soft limits when there's no reason to break them:

```
$this->logger->error(
  'Could not create a Hubspot API client. Error: @error',
  ['@error' => $e->getMessage()]
);
```

However, I don't think we want to catch the exception here. If there is an error and we can't create the client then there is no point in proceeding and considering the service object successfully constructed, we won't be able to sync anything.

We can let it fail here (remove `try/catch`), it will be the responsibility of the caller to decide what to do if there is an error. For example, the queue worker by default will faill processing an individual queue item and will proceed to the next.